sys/sched.c

```diff
@@ -59,8 +59,11 @@ void sched_switch(thread_t *newtd) {
 
   if (newtd == NULL) {
     newtd = runq_choose(&runq);
-    if (newtd)
+    if (newtd) {
+      if(newtd->td_type == TD_USER && newtd->user_map)
+          set_active_vm_map(newtd->user_map);
```

Comment:
Shouldn't we introduce process structure? This shouldn't be that hard. Besides I don't see how to change vm_map without calling vm_map_change in ctx_switch, which would require change from LEAF to NESTED, is this a problem, or you have better idea?